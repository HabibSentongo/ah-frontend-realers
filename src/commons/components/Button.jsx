import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const {
    buttonEvent, disabled, buttonClass, buttonName,
  } = props;

  return (
    <button
      className={buttonClass}
      type="button"
      onClick={buttonEvent}
      disabled={disabled}
    >
      {buttonName}
    </button>
  );
};

Button.defaultProps = {
  disabled: false,
};

Button.propTypes = {
  disabled: PropTypes.bool,
  buttonName: PropTypes.string.isRequired,
  buttonClass: PropTypes.string.isRequired,
  buttonEvent: PropTypes.func.isRequired,
};

export default Button;
