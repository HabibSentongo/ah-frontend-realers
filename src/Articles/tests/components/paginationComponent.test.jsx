import React from 'react';

import { mount } from 'enzyme';

import Pagination from '../../components/paginationComponent';
import Button from '../../../commons/components/Button';

describe('Pagination', () => {
  const props = {
    previous: null,
    next: 'next',
    paginate: jest.fn(),
    count: 20,
  };
  let wrapper = mount(<Pagination {...props} />);

  it('this should render without a problem', () => {
    expect(wrapper).toMatchSnapshot();
  });

  const prevButton = wrapper.find(Button).at(0);
  const nxtButton = wrapper.find(Button).at(1);

  it('should disable buttons if null', () => {
    expect(prevButton.props().disabled).toBe(true);
    expect(nxtButton.props().disabled).toBe(false);
  });


  it('should call the paginate function on button click', () => {
    const event = {
      target: {
        type: 'button',
        name: '',
      },
      preventDefault: jest.fn(),
    };

    nxtButton.simulate('click', event);
    expect(props.paginate).toBeCalled();
    prevButton.simulate('click', event);


    props.previous = 'previous';
    wrapper = mount(<Pagination {...props} />);
    wrapper.find(Button).at(0).simulate('click', event);
    expect(props.paginate).toBeCalled();
  });
});