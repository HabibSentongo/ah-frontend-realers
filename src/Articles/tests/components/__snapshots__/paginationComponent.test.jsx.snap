// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination this should render without a problem 1`] = `ReactWrapper {}`;
